src/main/java/frc/robot/systems/EveryBotIntakeFSM.java

```diff
@@ -0,0 +1,340 @@
+			return EveryBotIntakeFSMState.IDLE_STOP;
+		}
+		switch (currentState) {
+			case INTAKING:
+				if (needsReset && isMotorAllowed && toggleUpdate) {
+					timer.reset();
+					timer.start();
+					needsReset = false;
+				}
+				if (timer.hasElapsed(TIME_RESET_CURRENT)) {
+					currLogs[tick % AVERAGE_SIZE] = spinnerMotor.getOutputCurrent();
+					tick++;
+
+					double avgcone = 0;
+					for (int i = 0; i < AVERAGE_SIZE; i++) {
+						avgcone += currLogs[i];
+					}
+					avgcone /= AVERAGE_SIZE;
+					if (avgcone > CURRENT_THRESHOLD || !(input.isIntakeButtonPressed())) {
+						return EveryBotIntakeFSMState.IDLE_STOP;
+					}
+				}
+				return EveryBotIntakeFSMState.INTAKING;
+			case IDLE_STOP:
+				if (input.isOuttakeButtonPressed()
+					&& flipMotor.getEncoder().getPosition() > FLIP_THRESHOLD) {
+					return EveryBotIntakeFSMState.OUTTAKING;
+				} else if (input.isIntakeButtonPressed()
+					&& !(flipMotor.getEncoder().getPosition() > FLIP_THRESHOLD)) {
+					return EveryBotIntakeFSMState.INTAKING;
+				} else if (input.isFlipButtonPressed()) {
+					//&& arm.getEncoderCount() > BASE_THRESHOLD) {
+					if (flipMotor.getEncoder().getPosition() > FLIP_THRESHOLD) {
+						return EveryBotIntakeFSMState.IDLE_FLIPCOUNTERCLOCKWISE;
+					} else {
+						return EveryBotIntakeFSMState.IDLE_FLIPCLOCKWISE;
+					}
+				} else {
+					return EveryBotIntakeFSMState.IDLE_STOP;
+				}
+			case IDLE_FLIPCLOCKWISE:
+				if (flipMotor.getEncoder().getPosition() > FLIP_THRESHOLD
+					|| input.isFlipAbortButtonPressed()) {
+					return EveryBotIntakeFSMState.IDLE_STOP;
+				} else if (input.isFlipButtonPressed()) {
+					return EveryBotIntakeFSMState.IDLE_FLIPCOUNTERCLOCKWISE;
+				} else if (flipMotor.getEncoder().getPosition() < FLIP_THRESHOLD) {
+					return EveryBotIntakeFSMState.IDLE_FLIPCLOCKWISE;
+				}
+			case IDLE_FLIPCOUNTERCLOCKWISE:
+				if (flipMotor.getEncoder().getPosition() < 0
+					|| input.isFlipAbortButtonPressed()) {
+					return EveryBotIntakeFSMState.IDLE_STOP;
+				} else if (input.isFlipButtonPressed()) {
+					//&& arm.getEncoderCount() > BASE_THRESHOLD) {
+					return EveryBotIntakeFSMState.IDLE_FLIPCLOCKWISE;
+				} else {
+					return EveryBotIntakeFSMState.IDLE_FLIPCOUNTERCLOCKWISE;
+				}
+			case OUTTAKING:
+				if (input.isOuttakeButtonPressed()) {
+					return EveryBotIntakeFSMState.OUTTAKING;
+				} else {
+					return EveryBotIntakeFSMState.IDLE_STOP;
+				}
+			default:
+				throw new IllegalStateException("Invalid state: " + currentState.toString());
+		}
+	}
+
+	/* ------------------------ FSM state handlers ------------------------ */
+	/**
+	 * Handle behavior in states.
+	 */
+	private void handleIntakingState() {
+		spinnerMotor.set(INTAKE_SPEED);
+		flipMotor.set(0);
+	}
+	private void handleIdleStopState() {
+		spinnerMotor.set(KEEP_SPEED);
+		flipMotor.set(0);
+	}
+	private void handleFlipClockWiseState() {
+		flipMotor.set(-TURN_SPEED);
+		if (flipMotor.getEncoder().getPosition() >= FLIP_THRESHOLD) {
```

Comment:
Why arent we using PID for flipping 180 so our encoder position doesn't go too much past the flip threshold

Reply:
Added